include/vkgcDefs.h

```diff
@@ -86,6 +86,7 @@
 //  %Version History
 //  | %Version | Change Description                                                                                    |
 //  | -------- | ----------------------------------------------------------------------------------------------------- |
+//  |     53.0 | Add optimizationLevel to PipelineOptions
```

Comment:
We don't need to bump the major version on adding a new struct member, as long as the default of 0 (what you'd get if the client didn't know about it) was sensible. But I guess here 0 is not a sensible default, and the client needs to set it to 2 or 3 to get the behaviour that they got before.

Reply:
The default when someone create a build info object is 0.  The practice is to initialize the entire struct to 0.  However, as you mentioned in this case 0 is noopt, which is not the behavior that people will want.  That is why I changed the major version number.

I tried a few different ways to get the value to default to 2, but no success.

We could change the meaning of the option to be "0 is some compiler chosen default, and 1-4 map to 0-3 respectively".  However, I feel that is confusing to for the user.  Let me know if you feel this is acceptable.

Reply:
What was the conclusion of this? I am seeing the confusing behaviour where if I use amdllpc to compile an old dumped pipeline, it uses CodeGenOpt::Less instead of CodeGenOpt::Default.

I see you added something in initCompileInfo to initialise *.options.optimizationLevel to CodeGenOpt::Default, but all those defaults get overwritten in processInputPipeline by:
```
  compileInfo.compPipelineInfo = pipelineState->compPipelineInfo;
  compileInfo.gfxPipelineInfo = pipelineState->gfxPipelineInfo;
```
i.e. it's using vfx's defaults, which are presumably to 0-initialise the entire struct.